File: Daily/sbgo/Leetcode852-山脉数组的顶峰索引.go
package sbgo

func peakIndexInMountainArray(arr []int) int {
	l, r := 0, len(arr)-1
	for l < r {
		mid := l + (r-l)/2
		if arr[mid] > arr[mid+1] {
			r = mid
		} else {
			l = mid + 1
		}
	}
	return l
}

File: Leetcode/Rust/Leetcode852-山脉数组的顶峰索引.rs
impl Solution {
    pub fn peak_index_in_mountain_array(arr: Vec<i32>) -> i32 {
        let (mut l,mut r) = (0,arr.len() - 1);
        while l < r{
            let mid = l + (r - l) / 2;
            match arr[mid] > arr[mid + 1]{
                true => r = mid,
                false => l = mid + 1,
            }
        }
        l as i32
    }
}